src/charm.py

```diff
@@ -1099,7 +1105,9 @@ def _external_host(self) -> Optional[str]:
         if external_hostname := self.model.config.get("external_hostname"):
             return cast(str, external_hostname)
 
-        return _get_loadbalancer_status(namespace=self.model.name, service_name=self.app.name)
+        return _get_loadbalancer_status(
+            namespace=self.model.name, service_name=f"{self.app.name}-lb"
```

Comment:
this will break the charm and is a bit tricky cause `_get_loadbalancer_status` is called in `traefik_route` prior to initializing `service_patch` and `service_patch` itself depends on `traefik_route`.

Reply:
I don't follow. I meant something like:
```suggestion
            namespace=self.model.name, service_name=self.service_patch.service_name"
```

Reply:
So the `service_patch` object includes a param `_service_ports` that requires the `TraefikRouteProvider` and `Traefik` objects to be instantiated prior to `service_patch` and because `TraefikRouteProvider` needs to be aware of `_external_host` which itself calls the `_get_loadbalancer_status` it fails with `AttributeError: 'TraefikIngressCharm' object has no attribute 'service_patch'`

below would work to overcome this but i'm not sure tbh if this actually looks any better than just hard-coding it

```
        try:
            return _get_loadbalancer_status(
                namespace=self.model.name, service_name=self.service_patch.service_name
            )
        except AttributeError:
            return None
```